sno/working_copy/mysql.py

```diff
@@ -0,0 +1,327 @@
+            return TimestampType
+        else:
+            # Don't need to specify type information for other columns at present, since we just pass through the values.
+            return None
+
+    def _write_meta(self, sess, dataset):
+        """Write the title (as a comment) and the CRS. Other metadata is not stored in a PostGIS WC."""
+        self._write_meta_title(sess, dataset)
+        self._write_meta_crs(sess, dataset)
+
+    def _write_meta_title(self, sess, dataset):
+        """Write the dataset title as a comment on the table."""
+        sess.execute(
+            f"ALTER TABLE {self.table_identifier(dataset)} COMMENT = :comment",
+            {"comment": dataset.get_meta_item("title")},
+        )
+
+    def _write_meta_crs(self, sess, dataset):
+        """Populate the spatial_ref_sys table with data from this dataset."""
+        # TODO - MYSQL-PART-2: Actually store CRS, if this is possible.
+        pass
+
+    def delete_meta(self, dataset):
+        """Delete any metadata that is only needed by this dataset."""
+        # TODO - MYSQL-PART-2: Delete any extra metadata that is not stored in the table itself.
+        pass
+
+    def _create_spatial_index_post(self, sess, dataset):
+        # Only implemented as _create_spatial_index_post:
+        # It is more efficient to write the features first, then index them all in bulk.
+
+        # TODO - MYSQL-PART-2 - We can only create a spatial index if the geometry column is declared
+        # not-null, but a datasets V2 schema doesn't distinguish between NULL and NOT NULL columns.
+        # So we don't know if the user would rather have an index, or be able to store NULL values.
+        return  # Find a fix.
+
+        L = logging.getLogger(f"{self.__class__.__qualname__}._create_spatial_index")
+
+        geom_col = dataset.geom_column_name
+
+        L.debug("Creating spatial index for %s.%s", dataset.table_name, geom_col)
+        t0 = time.monotonic()
+
+        sess.execute(
+            f"ALTER TABLE {self.table_identifier(dataset)} ADD SPATIAL INDEX({self.quote(geom_col)})"
+        )
+
+        L.info("Created spatial index in %ss", time.monotonic() - t0)
+
+    def _drop_spatial_index(self, sess, dataset):
+        # MySQL deletes the spatial index automatically when the table is deleted.
+        pass
+
+    def _quoted_trigger_name(self, dataset, trigger_type):
+        trigger_name = f"sno_{dataset.table_name}_{trigger_type}"
+        return f"{self.DB_SCHEMA}.{self.quote(trigger_name)}"
+
+    def _create_triggers(self, sess, dataset):
+        table_identifier = self.table_identifier(dataset)
+        pk_column = self.quote(dataset.primary_key)
+
+        sess.execute(
+            text_with_inlined_params(
+                f"""
+                CREATE TRIGGER {self._quoted_trigger_name(dataset, 'ins')}
+                    AFTER INSERT ON {table_identifier}
+                FOR EACH ROW
+                    REPLACE INTO {self.SNO_TRACK} (table_name, pk)
```

Comment:
It must be auto-casting - nothing would work otherwise since our big-3 test data (points, polygons, table) all have integer primary keys, but tracking table is strings.